grease-cli/tests/sanity/pass/dwarf-shape-recursive/test.c

```diff
@@ -0,0 +1,19 @@
+
+struct llnode {
+  int x;
+  struct llnode *next;
+};
+
+int sum_list(struct llnode *in) {
+  struct llnode *curr = in;
+  int sum = 0;
+  while (curr) {
+    sum += curr->x;
+    curr = curr->next;
+  }
+  return sum;
+}
+
+// all: flags {"--symbol", "sum_list", "--enable-dwarf-preconditions", "--no-heuristics"}
+// x64: go(prog) 
+// x64: ok()
```

Comment:
Perhaps add some tests with different values for recursive type unrollings?

Reply:
Also FWIW it might be nice to directly test the generated shapes, along the lines of https://github.com/GaloisInc/grease/pull/254/files#diff-f49829d9043ff039dfb0971ac8c722f5efb92871f4a6ab08e91082b9dbaf2e90R109